### DISCUSSION TITLE
bug when filtering data

### DISCUSSION THREAD

**Discussion by dimasarfan01** [pmndrs/valtio#751] [Bug report] [answered]
so i've been following valtio getting started but i get an issue when changing a filter. i double checked again by copy paste in codesandbox but i got the same result

https://github.com/pmndrs/valtio/assets/77246142/8e08811b-3ba2-4442-b7c2-190c24b5074c

i'm using next.js ver. 13.4.8

**Comment by dai-shi** [comment]
Thanks for reporting.
Can you identify if the issue is next.js specific? Can you reproduce the bug with https://codesandbox.io/s/react-new ?

**Reply by dimasarfan01** [reply]
yes here 
https://codesandbox.io/p/sandbox/smoosh-shadow-s5sd2p (next)
https://codesandbox.io/s/heuristic-worker-w57wlv (react)

**Reply by dai-shi** [reply]
> https://codesandbox.io/s/heuristic-worker-w57wlv (react)

That's great to know. Can identify which valtio version causes this, or is it from the beginning?
Thanks for your investigation.

**Reply by dimasarfan01** [reply]
recently i've been trying again to ensure if it was because react or next js, so i do some test in getting started codesandbox (https://codesandbox.io/s/valtio-to-do-list-hm5kwz) here the result:


https://github.com/pmndrs/valtio/assets/77246142/56d23f33-8c53-4da9-beb9-d2de7345d160

im using "valtio": "^1.10.6"

**Reply by dai-shi** [reply]
Thanks for confirming. I tried downgrading to v1.0.0, but the behavior is the same.

I dug into it and it seems like a bug in the example. Here's the fix: https://codesandbox.io/s/valtio-to-do-list-forked-mw8kgp?file=/src/App.tsx

**Reply by dimasarfan01** [reply]
thanks for the fix, but when i tried it on next js the bug still occurs

https://codesandbox.io/p/sandbox/smoosh-shadow-s5sd2p

https://github.com/pmndrs/valtio/assets/77246142/00c4243c-0622-42c1-b683-0c3637943d55

**Reply by dai-shi** [reply, marked as answer]
Ooops, I forgot to fix `removeTodo`. Here's the new one: https://codesandbox.io/s/valtio-to-do-list-forked-6w9h3z?file=/src/App.tsx

**Reply by dai-shi** [reply]
https://github.com/pmndrs/valtio/pull/753 to fix the docs.
